src/libs/Middleware/Reauthentication.js

```diff
@@ -58,30 +58,18 @@ function Reauthentication(response, request, isFromSequentialQueue) {
                 // There are some API requests that should not be retried when there is an auth failure like
                 // creating and deleting logins. In those cases, they should handle the original response instead
                 // of the new response created by handleExpiredAuthToken.
-                const shouldRetry = lodashGet(request, 'data.shouldRetry');
+                // If the request was from the sequential queue we don't want to return, we want to run the reauthentication callback and retry the request
                 const apiRequestType = lodashGet(request, 'data.apiRequestType');
-
-                // For the SignInWithShortLivedAuthToken command, if the short token expires, the server returns a 407 error,
-                // and credentials are still empty at this time, which causes reauthenticate to throw an error (requireParameters),
-                // and the subsequent SaveResponseInOnyx also cannot be executed, so we need this parameter to skip the reauthentication logic.
+                const isDeprecatedRequest = !apiRequestType;
                 const skipReauthentication = lodashGet(request, 'data.skipReauthentication');
-                if ((!shouldRetry && !apiRequestType) || skipReauthentication) {
-                    if (isFromSequentialQueue) {
-                        return data;
-                    }
-
+                if (!isFromSequentialQueue && (isDeprecatedRequest || skipReauthentication)) {
                     if (request.resolve) {
                         request.resolve(data);
+                        return;
                     }
                     return data;
                 }
 
-                // We are already authenticating and using the DeprecatedAPI so we will replay the request
-                if (!apiRequestType && NetworkStore.isAuthenticating()) {
-                    MainQueue.replay(request);
-                    return data;
-                }
```

Comment:
**NAB / more clean up (that maybe we should just do now)**

Is it accurate to say at this point that anything below this _must_ be from the new API? I think so right? Because we return early if `!isFromSequentialQueue && isDeprecatedRequest`.

So all the stuff with the `MainQueue` after reauthenticate can be removed? Doesn't seem like there are any requests passing through the `MainQueue` at this point that actually need reauth so they can't get to those parts of the code?

Can we remove this?

https://github.com/Expensify/App/blob/1e6425131edf7d62ce703cb88f034ce6da2a8f4a/src/libs/Middleware/Reauthentication.js#L84

And this:

https://github.com/Expensify/App/blob/1e6425131edf7d62ce703cb88f034ce6da2a8f4a/src/libs/Middleware/Reauthentication.js#L93
?

Reply:
I'm not sure if we can remove these, let's investigate in a follow up?

Reply:
Sure. Maybe we can add some logs for it in another PR. I can do it. I'd like to get rid of any dead code if can.